a10_openstack_lib/resources/a10_certificate.py

```diff
@@ -0,0 +1,177 @@
+            'allow_post': False,
+            'allow_put': True,
+            'validate': {
+                'type:uuid': None
+            },
+            'is_visible': True,
+            'primary_key': True
+        },
+        'tenant_id': {
+            'allow_post': True,
+            'allow_put': False,
+            'required_by_policy': True,
+            'is_visible': True
+        },
+        'name': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None
+            },
+            'is_visible': True,
+            'default': ''
+        },
+        'description': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None
+            },
+            'is_visible': True,
+            'default': '',
+        },
+        'cert_data': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None
+            },
+            'is_visible': True,
+            'default': ''
+        },
+        'key_data': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None
+            },
+            'is_visible': True,
+            'default': ''
+        },
+        'intermediate_data': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None,
+            },
+            'is_visible': True,
+            'default': ''
+        },
+        'password': {
+            'allow_post': True,
+            'allow_put': True,
+            'validate': {
+                'type:string': None,
+            },
+            'is_visible': True,
+            'default': ''
+
+        }
+    },
+    CERTIFICATE_BINDINGS: {
+            'id': {
+            'allow_post': False,
+            'allow_put': False,
+            'is_visible': True,
+            'primary_key': True,
+            'validate': {
+                'type:uuid': None
+            }
+        },
+        'tenant_id': {
+            'allow_post': True,
+            'allow_put': False,
+            'required_by_policy': True,
+            'is_visible': True
+        },
+        'certificate_id': {
+            'allow_post': True,
+            'allow_put': True,
+            'is_visible': True
+        },
+        'listener_id': {
+            'allow_post': True,
+            'allow_put': True,
+            'is_visible': True
+        },
+        'listener_name': {
+            'allow_post': False,
+            'allow_put': False,
+            'is_visible': True
+        },
+        'certificate_name': {
+            'allow_post': False,
+            'allow_put': False,
+            'is_visible': True
+        }
+    }
+}
+
+
+def convert_to_lower(input):
+    try:
+        return input.lower()
+    except AttributeError:
+        return input
```

Comment:
Trivial: none of this conversion code is used, it should all be moved to one file.